### PR TITLE
TINY-6069

### PR DESCRIPTION
**Pull request by tinydylan**
Related Ticket:

Description of Changes:
* Remove obsolete demo page
* Fix demo page for TinyMCE 5

Pre-checks:
* [X] Changelog entry added (n/a)
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/` for new features (if applicable)
* [X] License headers added on new files (if applicable)

Review:
* [X] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/demo/ts/demo/TinyMceDemo.ts [hunk 1]
```diff
@@ -19,10 +19,10 @@ export default function () {
     // images_upload_credentials: true,
     skin_url: '../../../../js/tinymce/skins/ui/oxide',
     setup(ed) {
-      ed.addButton('demoButton', {
+      ed.ui.registry.addButton('demoButton', {
         type: 'button',
```
Comment (lnewson) [inline modules/tinymce/src/core/demo/ts/demo/TinyMceDemo.ts:23]: You could remove this if you want, as it does nothing in v5.

Comment (tinydylan) [reply modules/tinymce/src/core/demo/ts/demo/TinyMceDemo.ts]: Thanks! Done.

**Comment by tinydylan** [comment]
@lnewson @metricjs DeepScan is complaining of "2 new and 2 fixed issues"... when it's just the same issues have moved to a different line number. In any case, I've included a fix - does this look ok?

**Comment by lnewson** [comment]
> @lnewson @metricjs DeepScan was complaining of "2 new and 2 fixed issues"... when it's just the same issues have moved to a different line number. In any case, I've included a fix - does this look ok?

Yeah, DeepScan is silly like that unfortunately :disappointed:. Changes took fine to me :+1:

**Pull request merged**